joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/FieldAccessTests.scala

```diff
@@ -0,0 +1,70 @@
+package io.joern.go2cpg.passes.ast
+
+import io.joern.go2cpg.testfixtures.GoCodeToCpgSuite
+import io.joern.gosrc2cpg.astcreation.Defines
+import io.shiftleft.codepropertygraph.generated.Operators
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, FieldIdentifier, Identifier, Literal, MethodParameterIn}
+import io.shiftleft.semanticcpg.language.*
+
+class FieldAccessTests extends GoCodeToCpgSuite {
+
+  "simple field access test" should {
+    val cpg = code("""
+        |package main
+        |type Person struct {
+        |	fname string
+        |	lname string
+        |}
+        |func (person Person) fullName() string {
+        |	return person.fname + " " + person.lname
+        |}
+        |func main() {
+        |	var a Person = Person{fname: "Pandurang", lname: "Patil"}
+        |	var fn string = a.fname
+        |}
+        |
+        |func foo() {
+        |   var a Person = Person{fname: "Pandurang", lname: "Patil"}
+        |   println(a.fname)
+        |}
+        |""".stripMargin)
+
+    "External Field access test" in {
+      val List(fieldAccessCall) = cpg.method("main").astChildren.fieldAccess.l
+      fieldAccessCall.name shouldBe Operators.fieldAccess
+      // TODO: We need to find and set the right full name
```

Comment:
Field access methods are typically the best way to resolve the type once the AST is created: https://github.com/joernio/joern/blob/master/semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/nodemethods/FieldAccessMethods.scala

Reply:
@pandurangpatil will explain in the call